tests/destructor/tgcdestructors.nim

```diff
@@ -10,10 +10,9 @@ a: @[4, 2, 3]
 0
 30
 true
-41 41'''
+(allocCount: 123, deallocCount: 82)'''
```

Comment:
So we leak 41 objects? How come?

Reply:
I think we just alloc more often than dealloc due to string/seq resizing

Reply:
Yeah, I mentioned this on #nim already, and I'm not sure what is the best way to solve this.

This implementation keeps track of all allocs, not only allocs made by seq_v2/str_v2. There is still memory alive when the statistics are gathered and printed. I guess the only way to solve this is to wait until after Nim is all done, somewhere after NimMainInit()